observability-kit-agent/src/main/java/com/vaadin/extension/instrumentation/client/ClientInstrumentation.java

```diff
@@ -35,106 +40,46 @@ public ElementMatcher<TypeDescription> typeMatcher() {
-                                    Span childSpan = createSpan(tracer,
-                                            childSpanNode);
-                                    childSpan.end(
-                                            childSpanNode.get("endTimeUnixNano")
-                                                    .asLong(),
-                                            TimeUnit.NANOSECONDS);
-                                }
-                            }
-                        }
-
-                        parentSpan.end(
-                                parentSpanNode.get("endTimeUnixNano").asLong(),
-                                TimeUnit.NANOSECONDS);
-                    }
-
-                    if (parentSpanNodes.size() == 0) {
-                        for (Map.Entry<String, JsonNode> childEntry : childSpanNodes
-                                .entrySet()) {
-                            JsonNode childSpanNode = childEntry.getValue();
-
-                            Span childSpan = createSpan(tracer, childSpanNode);
-                            childSpan.end(childSpanNode.get("endTimeUnixNano")
-                                    .asLong(), TimeUnit.NANOSECONDS);
-                        }
-                    }
-                }
-            }
-        }
-
-        public static Span createSpan(Tracer tracer, JsonNode spanNode) {
-            SpanBuilder spanBuilder = tracer
-                    .spanBuilder("Client: " + spanNode.get("name").asText());
-            spanBuilder.setSpanKind(SpanKind.SERVER);
-            spanBuilder.setStartTimestamp(
-                    spanNode.get("startTimeUnixNano").asLong(),
-                    TimeUnit.NANOSECONDS);
-
-            Span span = spanBuilder.startSpan();
-            int status = spanNode.get("status").get("code").asInt();
-            span.setStatus(StatusCode.values()[status + 1]);
-
-            for (JsonNode attributeNode : spanNode.get("attributes")) {
-                if (attributeNode.get("value").has("stringValue")) {
-                    span.setAttribute(attributeNode.get("key").asText(),
-                            attributeNode.get("value").get("stringValue")
-                                    .asText());
-                } else if (attributeNode.get("value").has("intValue")) {
-                    span.setAttribute(attributeNode.get("key").asText(),
-                            attributeNode.get("value").get("intValue").asInt());
-                }
-            }
-
-            for (JsonNode eventNode : spanNode.get("events")) {
-                span.addEvent(eventNode.get("name").asText(),
-                        eventNode.get("timeUnixNano").asLong(),
-                        TimeUnit.NANOSECONDS);
+        transformer.applyTransformer((builder, typeDescription, classLoader,
+                module, protectionDomain) -> {
+            try {
+                Class<?> helperClazz = classLoader.loadClass(
```

Comment:
I wonder if it makes sense to add a comment here to explain why this reflection hack is needed.